fs/unpacker.go

```diff
@@ -77,7 +77,6 @@ func (lu *layerUnpacker) Unpack(ctx context.Context, desc ocispec.Descriptor, mo
 	if err != nil {
 		return fmt.Errorf("cannot fetch layer: %w", err)
 	}
-	defer rc.Close()
 
 	if !local {
 		if err = lu.fetcher.Store(ctx, desc, rc); err != nil {
```

Comment:
nit: we need to close `rc` on error here too.

Reply:
I don't think so, since it looks like all the possible return values in [Fetch](https://github.com/awslabs/soci-snapshotter/blob/335515f746f50c964ed48159257e1aeba04805b6/fs/artifact_fetcher.go#L88) only return nil upon returning an error, so this would actually segfault.

We could be extra safe, though, and just do something like

```golang
if rc != nil {
    rc.Close()
}
```

I don't think it's super necessary but if you think it's better we can.

Reply:
I think the weakness with that approach is it assumes the current implementation for a interface. My suggestion would be to keep the close on line 80 and add another close on line 90 after the `rc` variable is reinstantiated.